### ISSUE TITLE
Consistent support for Namespaces in Scope and Target on SuppressMessage

### ISSUE THREAD

**Issue by Lexcess** [DotNetAnalyzers/StyleCopAnalyzers#1628]
This is probably considered an enhancement (especially since old Stylecop was idiosyncratic on this as well) but the handling of SuppressMessageAttribute is inconsistent.

Some suppressions seem to work as expected on a namespace level (and in fact can be generated as such by a code fix). 
```c#
[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1652:Enable XML documentation output", Justification = "Not in this namespace.", Scope = "namespace", Target = "~N:Foo")]
```
Others such as the following do not: 
```c#
[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.SpacingRules", "SA1025:Code must not contain multiple whitespace in a row", Justification = "Not in this namespace.", Scope = "namespace", Target = "~N:Foo")]
```
I forgot this was a weakness of the old stylecop and spent a while on a recent project trying to find out what I did wrong. Having what looks like perfectly well constructed suppressions be ignored is frustrating. Googling around shows I'm not entirely alone in this going against expectations either.

**Comment by sharwell** [comment]
If anything, this would be a Roslyn bug. StyleCop Analyzers reports diagnostics it finds whether or not they are suppressed, and Roslyn determines which ones to show based on the presence (or absence) of `#pragma warning disable` lines and `[SuppressMessage]` attributes.

:memo: You should not disable [SA1652](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1652.md). See the documentation of that rule and #1629 for additional information.

**Comment by Lexcess** [comment]
Ah fair enough, I'll see about raising it there then. On the not disabling SA1652, I used it only as an example (and I think not the only one?), but it is currently one of the suggested code fix options.
